File: DotNETcapstone/DotNETcapstone/Models/RequestLine.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace DotNETcapstone.Models
{
    public class RequestLine
    {

        public int Id { get; set; }
        private static int NextId { get; set; }

       
        public virtual int RequestId { get; set; }
        [ForeignKey("RequestId")]
        public virtual Request Request { get; set; }

        public virtual int ProductId { get; set; }
        [ForeignKey("ProductId")]
        public virtual Product Product { get; set; }

        public int Quantity { get; set; }

        public RequestLine() {
            Id = ++NextId;
        }

    }
}

File: springjavacapstone/src/main/java/com/example/bootcamp/sales/requestline/Requestline.java
package com.example.bootcamp.sales.requestline;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import com.example.bootcamp.sales.product.Product;
import com.example.bootcamp.sales.request.Request;
import com.fasterxml.jackson.annotation.JsonBackReference;

 @Entity
 
public class Requestline {
		@Id	
		@GeneratedValue(strategy=GenerationType.IDENTITY)
		 public int id;
	     private int quantity;

	    
		@JsonBackReference
	    @ManyToOne(optional=false)
	    @JoinColumn(name="requestId")
	     private Request request;
		
	    
		@ManyToOne(optional=false)
		@JoinColumn(name="productId")
		private Product product;
		////////////////////////////////////////////////////
		public Requestline() {}
		////////////////////////////////////////////////////
		
		
		
		
		public int getId() {
			return id;
		}
		public void setId(int id) {
			this.id = id;
		}
		public int getQuantity() {
			return quantity;
		}
		public void setQuantity(int quantity) {
			this.quantity = quantity;
		}
		public Request getRequest() {
			return request;
		}
		public void setRequest(Request request) {
			this.request = request;
		}
		public Product getProduct() {
			return product;
		}
		public void setProduct(Product product) {
			this.product = product;
		}
		
		
		
	
}
